upgrade/upgrade-scripts/verify-impl

```diff
@@ -55,40 +144,12 @@ if $opt_d; then
 	VERIFY_LIVE_MDS_OPT="-disableConsistentMdsZfsDataUtil"
 fi
 
-#
-# We use this directory to store the upgrade verification logs of this
-# container. This directory is accessible from both the container, and
-# the host. This allows the software running on the host to easily
-# access the logs after the verification is performed.
-#
-mkdir -p "${LOG_DIRECTORY}/${CONTAINER}" ||
-	die "failed to create directory: '${LOG_DIRECTORY}/${CONTAINER}'"
-
 trap run_support_info EXIT
 
-/usr/bin/java \
-	-Dlog.dir=/var/delphix/server/upgrade-verify \
-	-Dmdsverify=true \
-	$VERIFY_DEBUG_OPT \
-	-jar /opt/delphix/server/lib/exec/upgrade-verify/upgrade-verify.jar \
-	-d "${opt_o:-${LOG_DIRECTORY}/${CONTAINER}/upgrade_verify.json}" \
-	-f "${opt_f:-1}" \
-	-l "${opt_l:-en-US}" \
-	-v "$IMAGE_VERSION" \
-	-pl 60 -ph 80 \
-	$VERIFY_LIVE_MDS_OPT ||
-	die "'upgrade-verify.jar' failed in verification container"
-
-#
-# This name is used by the "upgrade-verify.jar" just executed, so we
-# cannot change this value without also modifying that JAR.
-#
-MDS_SNAPNAME="MDS-CLONE-upgradeverify"
+verify_run_verify_jar
 
-/opt/delphix/server/bin/dx_manage_pg stop -s "$MDS_SNAPNAME" ||
-	die "failed to stop postgres"
-
-/opt/delphix/server/bin/dx_manage_pg cleanup -s "$MDS_SNAPNAME" ||
-	die "failed to cleanup postgres"
+if is_stack_only_upgrade; then
```

Comment:
If I recall correctly, reboot required upgrades have hard dependency between the app-stack and bits that are available once reboot happens, and because we are verifying before reboot has happened, we cannot verify stack stack in those cases.

Reply:
Yea, I see what you mean.

Currently, the incompatibility you're worried about doesn't exist, but I guess it could exist at some point in the future.

If it was me, I'd opt for the most simple solution, and just run stack startup in all cases, since practically it should work for all cases right now. Then, if startup didn't work for the "reboot required" case later, we could do something like what you have here.

I just feel like since we don't actually have an upgrade scenario where this is a problem, we likely won't get good test coverage of this exception case, so I'm worried the additional complexity will come without any additional benefit. And the complexity can regress in a later change, and we might not notice, since we won't have this code path regularly tested.

But, if you think it'd be better to do the conditional now, I'm OK with that too.

Reply:
At the very least, if we keep the conditional, I think we should add a comment that adequately explains the situation.. i.e. that we can't reliably start the new stack on the current kernel configuration for "reboot required" upgrades, so we skip that step here.

Reply:
This comment will be resolved as part of addressing comment about splitting `verify-impl` into `verify-jar` and `verify-stack`